### PR TITLE
fix: remove task prototype when resample

### PR DESCRIPTION
**Pull request by be-marc**

### CODE REVIEW

File: R/worker.R [hunk 1]
```diff
@@ -262,6 +262,12 @@ workhorse = function(iteration, task, learner, resampling, param_values = NULL,
     learner$state$model = NULL
   }
 
+  # repeated saving of the prototype leads to large ResultData objects if the task contains many columns, factor levels or attributes
+  learner$state$data_prototype = NULL
+  learner$state$task_prototype = NULL
+  learner$fallback$state$data_prototype = NULL
```
Comment (sebffischer) [inline R/worker.R:268]: check first whether fallback exists.

**Review by sebffischer — CHANGES_REQUESTED** [review]
Also I think we should: 

* [ ] Add the column names to the learner state, then after the new version of mlr3 is released, we can change the `ordered_features()` function from mlr3learners/ mlr3extralearners
* [ ] Start removing the unused data_prototype (?)

**Comment by be-marc** [comment]
> Start removing the unused data_prototype (?)

I will open an PR at vetiver.

File: R/worker.R (hunk 1 above)
Comment (be-marc) [reply R/worker.R:268]: There is no error even if the fallback does not exist.

**Comment by be-marc** [comment]
Blocked by https://github.com/mlr-org/mlr3learners/pull/278.

**Pull request merged**